File: 프로그래머스/lv0/120848. 팩토리얼/팩토리얼.swift
import Foundation

func solution(_ n:Int) -> Int {
    var result = 0
    var idx = 0
    while result <= n {
        idx += 1
        result = factorial(idx)
    }
    return idx - 1
}

func factorial(_ n: Int) -> Int {
    if n == 1 {
        return 1
    }
    return factorial(n - 1) * n
}

File: 프로그래머스/lv0/120848. 팩토리얼/팩토리얼.js
function solution(n) {
    const factorial = (num) => {
        if (num === 1) return 1;
        return factorial(num - 1) * num;
    };
    let answer = 1;
    while (factorial(answer) <= n) {
        answer += 1;
    }
    return answer - 1;
}